### PR TITLE
Add auto gpt agent

### PR DESCRIPTION
**Pull request by slavakurilyak**
- Add an experimental version of `agentx/agents/autogpt`
- Modify the experimental version of `agentx/agents/babyagi`
- Modify `agentx/memory/faiss.py` to work with the experimental version of `agentx/agents/autogpt`
- Add tests for `agentx/agents/autogpt`
- Update usage instructions for README

### CODE REVIEW

File: agentx/agents/babyagi/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+# agentx/agents/babyagi/__init__.py
```
Comment (rG2ee) [inline agentx/agents/babyagi/__init__.py:1]: Why did you add comments on the first line for all files?

**Comment by rG2ee** [comment]
Also why did you apply changes to babyagi? This PR is related to the auto gpt integration. We can update babyagi in another pr.

**Comment by rG2ee** [comment]
Also you added `agentx/agents/autogpt/main.py` and `agentx/auto_gpt.py` which basically do the same. Please merge the two files. 
The objective is predefined. Is this indented?

**Comment by rG2ee** [comment]
Please add a simple test case like
- Give me the result on 11 + seven
- What's the age of leonardo dicaprios girlfriend?

**Comment by slavakurilyak** [comment]
> Also you added agentx/agents/autogpt/main.py and agentx/auto_gpt.py which basically do the same. Please merge the two files.

All scripts in `agentx` are part of the 'stable version' which uses a single-file structure. All scripts in `agentx/agents` are part of the 'experimental version' which uses a modular file (multi-file) structure.

**Comment by rG2ee** [comment]
It's really bad practise to add the file location on top each corresponding file. When using a git loader 
(https://github.com/hwchase17/langchain/blob/master/langchain/document_loaders/git.py) , one get a list of Documents (https://github.com/hwchase17/langchain/blob/master/langchain/schema.py#L304)
either the file path information is given there ore one patches the loader. 
Placing it in each file is a bad approach.

**Comment by rG2ee** [comment]
> All scripts in agentx are part of the 'stable version' which uses a single-file structure. All scripts in agentx/agents are part of the 'experimental version' which uses a modular file (multi-file) structure.

Then we should edit the file structure to

agentx/
--experimenal/
  --babyagi.py
  --autogpt.py
autogpt.py (production code)  
babyagi.py (production code)  
...

**Comment by slavakurilyak** [comment]
> > All scripts in agentx are part of the 'stable version' which uses a single-file structure. All scripts in agentx/agents are part of the 'experimental version' which uses a modular file (multi-file) structure.
> 
> Then we should edit the file structure to
> 
> agentx/ --experimenal/ --babyagi.py --autogpt.py autogpt.py (production code) babyagi.py (production code) ...

This can be accomplished with a new pull request

**Comment by slavakurilyak** [comment]
> It's really bad practise to add the file location on top each corresponding file. When using a git loader (https://github.com/hwchase17/langchain/blob/master/langchain/document_loaders/git.py) , one get a list of Documents (https://github.com/hwchase17/langchain/blob/master/langchain/schema.py#L304) either the file path information is given there ore one patches the loader. Placing it in each file is a bad approach.

I don't see it this way. Teaching file paths to LLMs is one of the better practices. Microsoft's [MM-ReAct](https://multimodal-react.github.io/) used "the file path as the input to ChatGPT. The file path functions as a placeholder, allowing ChatGPT to treat it as a black box."

In my experiments with ChatGPT, I found more accurate results when I was teaching the LLM about file and folder structure using trees and relative file paths.

While LangChain already has a git loader, not every developer is using this library when improving their codebase.

**Comment by rG2ee** [comment]
Looks good. Feel free to merge.
